### PR TITLE
feat: add get_budget to NWCClient

### PR DESCRIPTION
**Pull request by rolznz**
Fixes https://github.com/getAlby/js-sdk/issues/261

### CODE REVIEW

File: src/NWCClient.ts [hunk 1]
```diff
@@ -47,6 +48,16 @@ export type Nip47GetInfoResponse = {
   notifications?: Nip47NotificationType[];
 };
 
+export type Nip47GetBudgetResponse =
+  | {
+      used_budget: number; // msats
+      total_budget: number; // msats
+      renews_at?: number; // timestamp
```
Comment (im-adithya) [inline src/NWCClient.ts:55]: This is defined as `number | undefined` but if `renewal_period` is never we get `renews_at: null` 

Screenshot 2024-10-22 at 1 20 02 PM

**Review by im-adithya — APPROVED** [review]
Just a small nit, otherwise tACK

File: src/NWCClient.ts (hunk 1 above)
Comment (rolznz) [reply src/NWCClient.ts:55]: Good catch, I think we should fix this in Alby Hub to omit `renews_at` in the json.

Comment (rolznz) [reply src/NWCClient.ts:55]: Fixed here: https://github.com/getAlby/hub/pull/726/commits/634e54d778bee73c0c4f73dceffa7d05b7a1d21c

Comment (im-adithya) [reply src/NWCClient.ts:55]: Resolved by https://github.com/getAlby/hub/pull/726/commits/634e54d778bee73c0c4f73dceffa7d05b7a1d21c

**Pull request merged**